cylc/flow/scheduler.py

```diff
@@ -886,6 +886,13 @@ def _load_pool_from_db(self):
         self.workflow_db_mgr.pri_dao.select_abs_outputs_for_restart(
             self.pool.load_abs_outputs_for_restart)
 
+        # Compute and release runahead tasks once after loading all tasks from
+        # the DB. This also causes spawning of parentless tasks out to the
+        # runahead limit, which may be necessary here if the stop point or
+        # runahead limit was changed for the restart.
+        self.pool.compute_runahead()
+        self.pool.release_runahead_tasks()
```

Comment:
Fair enough, it didn't occur to me to move these lines here instead of inside `load_db_task_pool_for_restart()`